src/test/scala/com/target/data_validator/validator/ColumnSumCheckSpec.scala

```diff
@@ -0,0 +1,335 @@
+          ColumnSumCheck("foo", maxValue = Some(int_8._2), inclusive = Some(Json.fromBoolean(true))))
+        val df = mkDf(spark = spark, "foo" -> List(6))
+        assert(!sut.configCheck(df))
+      }
+    }
+
+    describe("functionality") {
+
+      val detailedErrors = false
+      val longListWithSum6 = "price" -> List(1L, 2L, 3L)
+
+      it("lower bound success") {
+        val check = ColumnSumCheck(
+          "price",
+          minValue = Some(Json.fromLong(0L))
+        )
+        val df = mkDf(spark, longListWithSum6)
+        val sut = ValidatorDataFrame(df, None, None, List(check))
+        val config = ValidatorConfig(1, 1, None, detailedErrors, None, None, List.empty)
+        assert(!sut.quickChecks(spark, mkDict())(config))
+        assert(!sut.failed)
+      }
+
+      it("lower bound failure") {
+        val check = ColumnSumCheck(
+          "price",
+          minValue = Some(Json.fromLong(6L))
+        )
+        val df = mkDf(spark, longListWithSum6)
+        val sut = ValidatorDataFrame(df, None, None, List(check))
+        val config = ValidatorConfig(1, 1, None, detailedErrors, None, None, List.empty)
+        assert(sut.quickChecks(spark, mkDict())(config))
+        assert(sut.failed)
+      }
+
+      it("lower bound inclusive success") {
+        val check = ColumnSumCheck(
+          "price",
+          minValue = Some(Json.fromLong(6L)),
+          inclusive = Some(Json.fromBoolean(true))
+        )
+        val df = mkDf(spark, longListWithSum6)
+        val sut = ValidatorDataFrame(df, None, None, List(check))
+        val config = ValidatorConfig(1, 1, None, detailedErrors, None, None, List.empty)
+        assert(!sut.quickChecks(spark, mkDict())(config))
+        assert(!sut.failed)
+      }
+
+      it("upper bound success") {
+        val check = ColumnSumCheck(
+          "price",
+          maxValue = Some(Json.fromLong(10L))
+        )
+        val df = mkDf(spark, longListWithSum6)
+        val sut = ValidatorDataFrame(df, None, None, List(check))
+        val config = ValidatorConfig(1, 1, None, detailedErrors, None, None, List.empty)
+        assert(!sut.quickChecks(spark, mkDict())(config))
+        assert(!sut.failed)
+      }
+
+      it("upper bound failure") {
+        val check = ColumnSumCheck(
+          "price",
+          maxValue = Some(Json.fromLong(6L))
+        )
+        val df = mkDf(spark, longListWithSum6)
+        val sut = ValidatorDataFrame(df, None, None, List(check))
+        val config = ValidatorConfig(1, 1, None, detailedErrors, None, None, List.empty)
+        assert(sut.quickChecks(spark, mkDict())(config))
+        assert(sut.failed)
+      }
+
+      it("upper bound inclusive success") {
+        val check = ColumnSumCheck(
+          "price",
+          maxValue = Some(Json.fromLong(6L)),
+          inclusive = Some(Json.fromBoolean(true))
+        )
+        val df = mkDf(spark, longListWithSum6)
+        val sut = ValidatorDataFrame(df, None, None, List(check))
```

Comment:
Not a fan of the `SumOfNumericColumnCheckBasicSetup` trait with the `testDfWithChecks` method?

Reply:
Not a fan is a bit too critical... There is a lot of opportunity for code reuse in the tests so I appreciate that, but there is a balance between that and making the tests readable in my opinion. I found myself scrolling up and down a lot to read the tests.  I'd like to refactor this a bit more, but I don't have a clear vision at the moment.